app/latigo/executor/__init__.py

```diff
@@ -133,132 +120,83 @@ def _perform_auth_checks(self):
-                revision=revision,
-                model_training_period=model_training_period,
+                task=task, revision=revision, model_training_period=model_training_period,
             )
-        except NoTagDataInDataLake as e:
-            # there's no data in the Data Lake for particular tag for given period of time.
-            logger.error(str(e), exc_info=True)
-            raise e
-        except Exception as e:
-            logger.error(f"Could not execute prediction for task '{task.project_name}.{task.model_name}': {e}")
-            raise e
-            # traceback.print_exc()
+        except IOC_DATA_EXCEPTIONS as e:
+            logger.warning(self.format_error_message(IOC_ERROR_IDENTIFIER, e=e, task=task))
+            return
+        except GORDO_EXCEPTIONS as e:
+            logger.warning(self.format_error_message(GORDO_ERROR_IDENTIFIER, e=e, task=task))
+            return
+
         return prediction_data
 
-    def _store_prediction_data_and_metadata(self, task: Task, prediction_data: PredictionDataSet):
+    def store_prediction_data_and_metadata(self, task: Task, prediction_data: PredictionDataSet):
         """Store prediction data (what represents the result of performing predictions on sensor data) and its metadata.
 
         1. Store just one bulk of prediction data. Do not send predictions for different models at ones.
         2. Store metadata of the prediction.
         """
+        # store predictions
         try:
-            # store predictions
             output_tag_names, output_time_series_ids = self.prediction_storage_provider.put_prediction(prediction_data)
-            self._log_task_execution_time(label="stored to TS API", chars_to_append="")
+            self._log_task_execution_time(label="stored to TS API")
+        except NoCommonAssetFound as e:
+            # raise e
+            logger.error(self.format_error_message("Prediction was not stored", e=e, task=task))
+            return
 
-            # store predictions metadata
-            input_time_series_ids = self._get_tags_time_series_ids_for_model(prediction_data)
+        # store predictions metadata
+        input_time_series_ids = self._get_tags_time_series_ids_for_model(prediction_data)
+        try:
             self.prediction_metadata_storage_provider.put_prediction_metadata(
                 prediction_data, output_tag_names, output_time_series_ids, input_time_series_ids
             )
-            self._log_task_execution_time(label="stored to Metadata API", chars_to_append="")
-
-        except Exception as e:
-            logger.error(
-                f"Could not store prediction or metadata for task {self.make_prediction_task_info(task)}. Error: {e}."
-            )
-            raise e
+            self._log_task_execution_time(label="stored to Metadata API")
+        except MetadataStoringError as e:
+            logger.error(self.format_error_message("Metadata was not stored", e=e, task=task))
```

Comment:
Why not keep it simple, like so:

```suggestion
            logger.exception("Metadata was not stored for task %s", task)
```

Reply:
to keep particular logs the same structure (error should be logged as well). This is need for the following visualization on the dashboard and monitor alerts. And if we change anything it would be done in one place. Latigo struggles with unified logging thus debugging might be a nightmare.

Reply:
There are better ways to support this:

1. Formatters: https://docs.python.org/3.7/library/logging.html#formatter-objects
2. Passing contextual information: https://docs.python.org/3/howto/logging-cookbook.html#adding-contextual-information-to-your-logging-output
3. Third-party libraries for additional context support, like this one: https://github.com/peopledoc/pylogctx